### DISCUSSION TITLE
How do you handle submodules in dream2nix projects?

### DISCUSSION THREAD

**Discussion by Avi-D-coder** [nix-community/dream2nix#514] [Q&A]
I'm trying dream2nix on this [repo's Rust](https://github.com/Avi-D-coder/cozo/tree/dream2nix).
The repo has a submodule `cozorocks/rocksdb`. What is the right approach using the submodule in d2n builds?
What about when the submodule is in a cargo git dependency?

**Comment by DavHau** [comment]
I think either the builtin or nixpkgs git fetcher to support fetching submodules.
So how about just fetching the repo including submodules, then throwing it at dream2nix?

**Reply by Avi-D-coder** [reply]
If I have a Rust crate that depends on another crate as a git dependency that includes a submodule, then dream2nix is fetching the dependency, it's not a flake input. I'm not clear how I could explicitly fetch a crate dependency, and tell dream2nix where to find that source for the sub dependency?
